src/core/src/FabricMaterialDefinition.cpp

```diff
@@ -83,8 +90,17 @@ const std::vector<MetadataUtil::PropertyDefinition>& FabricMaterialDefinition::g
 
 // In C++ 20 we can use the default equality comparison (= default)
 bool FabricMaterialDefinition::operator==(const FabricMaterialDefinition& other) const {
+    bool overlaysMatch = getImageryOverlayTypes().size() == other.getImageryOverlayTypes().size();
+    if (overlaysMatch) {
+        for (size_t i = 0; i < getImageryOverlayTypes().size(); i++) {
+            if (getImageryOverlayTypes()[i] != other.getImageryOverlayTypes()[i])
+                overlaysMatch = false;
+            continue;
+        }
+    }
+
     return _hasVertexColors == other._hasVertexColors && _hasBaseColorTexture == other._hasBaseColorTexture &&
-           _featureIdTypes == other._featureIdTypes && _imageryLayerCount == other._imageryLayerCount &&
+           _featureIdTypes == other._featureIdTypes && overlaysMatch &&
            _tilesetMaterialPath == other._tilesetMaterialPath && _properties == other._properties;
 }
```

Comment:
You should be able to just do `_imageryOverlayRenderPipes == other._imageryOverlayRenderPipes` (assuming that `_imageryOverlayTypes` is removed)

Reply:
Nevermind the last comment. You spotted the issue in the equality operator.